R/utilities-qualification.R

```diff
@@ -202,3 +202,65 @@ getOutputsFromGOFMergedPlotsConfiguration <- function(plot) {
 # Vector of allowable plot types in configuration plan.
 # TODO deprecate vector using enum ConfigurationPlots
 QualificationPlotTypes <- c("GOFMergedPlots", "ComparisonTimeProfilePlots", "DDIRatioPlots", "TimeProfile")
+
+
+
+
+#' @title separateVariableFromUnit
+#' @description Split a string containing a varialbe and unit description into a list of two strings that separates the variable and the unit.
+#' @param variableUnitString is a string that has the format 'Variable [unit]' or 'Variable'
+#' @return A named list, with fields 'name' and 'unit'.  If variableUnitString lacks a '[unit]' substring, then the unit field is returned as an empty string.
+separateVariableFromUnit <- function(variableUnitString) {
+  splitVariableUnitString <- strsplit(x = sub("[ []", replacement = "", x = variableUnitString), split = "[][]")[[1]]
+  return(list(
+    name = trimws(splitVariableUnitString[1]),
+    unit = ifelse(test = is.na(x = splitVariableUnitString[2]), yes = "", no = splitVariableUnitString[2])
+  ))
+}
+
+
+#' @title parseObservationsDataFrame
+#' @description Function to read the variable names and units in the first two columns of an observations data frame used for qualification. First column stores timepoints.  Second column stores measurements of a quantity corresponding to timepoints in first column.
+#' @param observationsDataFram
+#' @return A named list with `time` and `output` fields.  Each field contains a list that is output by `separateVariableFromUnit` with fields that store the variable name and the unit.
+parseObservationsDataFrame <- function(observationsDataFrame) {
+  namesObservationsDataFrame <- names(observationsDataFrame)
+  return(list(
+    time = separateVariableFromUnit(namesObservationsDataFrame[1]),
+    output = separateVariableFromUnit(namesObservationsDataFrame[2])
+  ))
+}
+
+
+
+#' @title massMoleConversion
+#' @description Function to map `Concentration (mass)` or `Mass` dimensions to `Concentration (molar)` and `Amount` respectively.
+#' @param dimension is string that is from among the valid OSP dimensions
+#' @return If `dimension` is `Concentration (mass)` or `Mass`, then return `Concentration (molar)` or `Amount` respectively, otherwise return `dimension`.
+massMoleConversion <- function(dimension) {
+  massMoleConversionList <- list()
+  massMoleConversionList[[ospDimensions$Mass]] <- ospDimensions$Amount
+  massMoleConversionList[[ospDimensions$`Concentration (mass)`]] <- ospDimensions$`Concentration (molar)`

```

Comment:
In case, to prevent bugs of namespace and such, could you use `ospsuite::ospDimensions` ?

Reply:
Added